bip78/src/sender/mod.rs

```diff
@@ -329,15 +330,15 @@ fn check_single_payee(psbt: &Psbt, script_pubkey: &Script, amount: bitcoin::Amou
 }
 
 fn clear_unneeded_fields(psbt: &mut Psbt) {
```

Comment:
What do you think of implementing this on `Psbt` and remove the `*_mut` methods?

especially `inputs_mut` and `outputs_mut` are dangerous because they can invalidate the checks